geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+	Kind *Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	Name ObjectName `json:"name"`
+
+	// Namespace is the namespace of the referenced object. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Support: Core
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+}
+```
+
+#### YAML
+
+The following example shows a `Gateway` with an HTTP listener and two child HTTPS `ListenerSets` with unique hostnames and certificates. 
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+---
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: ListenerSet
+metadata:
+  name: first-workload-listeners
+  labels:
+    gateway.networking.k8s.io/gateway.name: parent-gateway
+spec:
+  parentRef:
+    name: parent-gateway
+    kind: Gateway
+    group: gateway.networking.k8s.io
+  listeners:
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+---
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: ListenerSet
+metadata:
+  name: second-workload-listeners
+spec:
+  parentRef:
+    name: parent-gateway
+    kind: Gateway
+    group: gateway.networking.k8s.io
+  listeners:
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+### Semantics
+
+#### Gateway <> ListenerSet Handshake
+
+By default a `Gateway` will allow `ListenerSets` in the same namespace to be attached. Users can prevent this behaviour by configuring their `Gateway` to disallow any listener attachment:
+
+```
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  allowedListeners:
+  - from: None
+```
+
+#### Listeners
+
+####  `ListenerSet.spec.parentRef` is immutable
+
+A `ListenerSet` `MUST` be created with a `parentRef` Updating the `parentRef` of a ListenerSet `MUST` be rejected by the API.
+
+#### `ListenerSet` Default Labels
+
+In order for clients of the API to easily discover `Gateways` and their `ListenerSets` we propose a set of labels to be defaulted when creating `ListenerSets`. This only applies if the `parentRef.kind` is a `Gateway`
+
+- `gateway.networking.k8s.io/gateway.name` is equal to `spec.parentRef.name`
+- `gateway.networking.k8s.io/gateway.namespace` is equal to `spec.parentRef.namespace`
+
+This allows clients to perform API [lookups](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering) using label selectors.
+
+TBD: Can conditional defaulting be done with CEL?
```

Comment:
Unfortunately I don't think so, but it would be incredibly helpful. @jpbetz or @cici37 do you know if there are any issues tracking something like this?

Reply:
Don't need this anymore - since we dropped the automatic gateway label